domain/src/Item.js

```diff
@@ -21,20 +21,39 @@ function calcAps(item) {
 
 function calcDps(item) {
     let res = (((item.minDmg + item.maxDmg) / 2) * item.aps);
-    if (item.critChance && item.critDmg) {
+    if (item.critChance > 0 && item.critDmg > 0) {
         if (item.critChance > 100) item.critChance = 100;
-        return (res + (res * (item.critChance / 100) * (item.critDmg / 100)));
+        res += (res * (item.critChance / 100) * (item.critDmg / 100));
     }
-    return res;
+    return res.toFixed(2);
 }
 
 function calcScore(item) {
```

Comment:
What the f? You already store `item.type` in db, don't you? And there is no such thing as `item.dps`. You have `calcDps` instead.